### DISCUSSION TITLE
Windows is using PowerShell Core instead of PowerShell as documented

### DISCUSSION THREAD

**Discussion by paresy** [community/community#26868] [Actions]
<p>Hi!</p>
<p>I have another thing that confuses me: It seems Actions uses PowerShell Core and not the default Desktop PowerShell.</p>
<p>My workflow action:</p>
<pre><code>name: Build (Windows)

on: [deployment]

jobs:
  build-win:

    runs-on: windows-latest

    if: github.event.deployment.environment == 'win'

    steps:
    - name: Checkout
      uses: actions/checkout@v1

    - name: Build
      run: $PSVersionTable.PSVersion
</code></pre>
<p>The output:</p>
<pre><code>Major Minor Patch PreReleaseLabel BuildLabel
----- ----- ----- --------------- ----------
6 2 3
</code></pre>
<p>As far as i know this refers to PoweShell 6, which is PowerShell Core.</p>
<p>The default PowerShell on Windows Server 2019 is PowerShell 5.1.</p>
<p>The documentation says: </p>
<pre><code>This is the default shell used on Windows. The Desktop PowerShell. GitHub appends the extension .ps1 to your script name.
</code></pre>
<p>This seems like a bug.</p>
<p>Regards,</p>
<p>Michael</p>

**Comment by weide-zhou** [comment]
<p>Hi,</p>
<p>Thank you for taking time to report this!</p>
<p>Powershell 6.2.3 is the correct version which will be invoked, please refer to my screenshot below.</p>
<p>The twitter link: <a href="https://twitter.com/GitHubHelp/status/1186648849995177985" rel="nofollow noopener">https://twitter.com/GitHubHelp/status/1186648849995177985</a></p>
<p>I also checked the github runner env, only powershell 6.2.3 is installed.</p>
<p><a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/software-in-virtual-environments-for-github-actions" rel="nofollow noopener">https://help.github.com/en/github/automating-your-workflow-with-github-actions/software-in-virtual-environments-for-github-actions</a></p>
<p> Thanks!</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/82fce3e60a510d434762d067e02045c1777718a3.png" data-download-href="/uploads/short-url/iGLSVGDruBIwQfzutbpdyHxvIWL.png?dl=1" title="Twitter.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181100373-9fcc5fde-e303-4b71-8228-c127f3125857.png" alt="Twitter.png" data-base62-sha1="iGLSVGDruBIwQfzutbpdyHxvIWL" width="538" height="499" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/82fce3e60a510d434762d067e02045c1777718a3_2_538x499.png, https://aws1.discourse-cdn.com/github/original/2X/8/82fce3e60a510d434762d067e02045c1777718a3.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/82fce3e60a510d434762d067e02045c1777718a3.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/82fce3e60a510d434762d067e02045c1777718a3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Twitter.png</span><span class="informations">712×661 58.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by paresy** [comment]
<p>Thanks for your anwser!</p>
<p>Am i correct, that the output for using shell: powerstell and shell: pwsh is the same on Windows?</p>
<p>If shell: powershell also calls Powershell Core it would be safe to remove the shell: powershell completely and just make shell: pwsh the default.</p>
<p>AFAIK Poweshell 5.x is installed by default on Windows Server 2019. Therefore the virtual env docs might not have reflected this properly?</p>
<p>Anyway it seems confusing to have two shell options which call the same PowerShell. The docs should clearly state why this choice has been made and when i should use shell: powershell and when to use shell: pwsh. At the moment i am not confident to make a proper choice.</p>
<p>(When reading the docs it seems that the difference should have been PowerShell Core vs PowerShell Desktop. If this difference was purposely then this is indeed a bug and shell: powershell should properly call the shipped PowerShell which is 5.x)</p>
<p>Regards,</p>
<p>Michael</p>

**Comment by paresy** [comment]
<p>It did some more research on this. I have a stock Windows Server 2019 installation.</p>
<p>Running in the Desktop PowerShell:</p>
<pre><code>PS C:\Users\Administrator&gt; $PSVersionTable

Name Value
---- -----
PSVersion 5.1.17763.592
PSEdition Desktop
PSCompatibleVersions {1.0, 2.0, 3.0, 4.0...}
BuildVersion 10.0.17763.592
CLRVersion 4.0.30319.42000
WSManStackVersion 3.0
PSRemotingProtocolVersion 2.3
SerializationVersion 1.1.0.1
</code></pre>
<p>Running in the additionally installed PowerShell Core:</p>
<pre><code>PS C:\Users\Administrator\build\win&gt; $PSVersionTable

Name Value
---- -----
PSVersion 6.2.3
PSEdition Core
GitCommitId 6.2.3
OS Microsoft Windows 10.0.17763
Platform Win32NT
PSCompatibleVersions {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion 2.3
SerializationVersion 1.1.0.1
WSManStackVersion 3.0
</code></pre>
<p>The documentation states the following: (<a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions" rel="nofollow noopener">https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions</a>)</p>
<pre><code>This is the default shell used on Windows. The Desktop PowerShell. GitHub appends the extension .ps1 to your script name.
</code></pre>
<p>The Desktiop edition is clearly the one with version 5.1.x. Therefore this seems like either a totaly wrong documentation (and as suggested a redundant shell: powershell option) or a bug that the wrong PowerShell is called.</p>
<p>Regards,</p>
<p>Michael</p>

**Comment by paresy** [comment]
<p>It is definitly a bug. </p>
<p>Using a custom shell as specified in the documentation will properly run the stock Desktop Powershell.</p>
<p>My modified workflow:</p>
<pre><code>- name: Build
      shell: powershell -command "&amp; '{0}'"
      run: |
        cd win
        .\build.ps1 -BuildConfiguration Release
</code></pre>
<p>Here is the output</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a8d41c16547fb4ba1d76f521924ff2dc94b6f9ac.png" data-download-href="/uploads/short-url/o5wENn8HS1X9c5xxrlBHf6FmLve.png?dl=1" title="debug_custom.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/260376/181100377-bf305e2b-3776-482b-84e7-c5d6619566bf.png" alt="debug_custom.png" data-base62-sha1="o5wENn8HS1X9c5xxrlBHf6FmLve" width="690" height="157" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a8d41c16547fb4ba1d76f521924ff2dc94b6f9ac_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">debug_custom.png</span><span class="informations">1022×234 7.72 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Using the default shell will run the PowerShell Core.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/fbd9ab9e495dec81c18956e4cec6c1faf064d72f.png" data-download-href="/uploads/short-url/zVYcvrOZC3cjBCi8GI1R80DYYSr.png?dl=1" title="debug_default.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/260376/181100378-15904232-9a98-44e3-a836-2acc77f5717a.png" alt="debug_default.png" data-base62-sha1="zVYcvrOZC3cjBCi8GI1R80DYYSr" width="690" height="210" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/fbd9ab9e495dec81c18956e4cec6c1faf064d72f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">debug_default.png</span><span class="informations">877×268 8.55 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Regards,</p>
<p>Michael</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Thanks for your feedback! There are two parameters here: pwsh / powershell,</p>
<p>for “shell: pwsh”: powershell core(6.2.3) will be invoked.</p>
<p>for “shell: powershell”: desktop powershell(5.1) will be invoked.</p>
<p>If you didn’t point out the shell parameter, powershell core will be invoked as it’s default.</p>
<p>As <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference" rel="nofollow noopener">offical doc</a> says:</p>
<p>Users can always opt out by not using the built-in shell, and providing a custom shell option like: pwsh -File {0}, or powershell -Command “&amp; ‘{0}’”, depending on need.</p>
<p>Thanks.</p>

**Comment by paresy** [comment]
<p>Thanks for the clarification. So it is just a simple documentation issue, which states that PowerShell Desktop is the default instead of PowerShell Core.</p>
<p>Regards,</p>
<p>Michael</p>

**Comment by paresy** [comment]
<p>I have fallen for this one again. Could you fix the documentation and correctly state that Powershell Core is the default shell on Windows?</p>
